File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
    public int MaxArea(int[] heights) {
        var result = int.MinValue;
        var start = 0;
        var end = heights.Length - 1;

        while(start < end)
        {
            var area = Math.Min(heights[start], heights[end]) * (end - start);

            result = Math.Max(result, area);

            if(heights[start] <  heights[end])
            {
                start++;
            }
            else
            {
                end--;
            }
        }

        return result;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        start, result = 0, 0
        s_t = [0] * 26

        for end in range(len(s)):
            s_t[ord(s[end]) - ord('A')] += 1

            while (end - start + 1) - max(s_t) > k:
                s_t[ord(s[start]) - ord('A')] -= 1
                start += 1
            
            result = max(result, end - start + 1)
        
        return result